pkg/blockbuilder/blockbuilder.go

```diff
@@ -273,30 +333,47 @@ func (b *BlockBuilder) consumePartitionSection(
 			"last_consumed_offset", lastConsumedOffset, "num_blocks", len(blockMetas))
 	}(time.Now())
 
+	b.kafkaClient.ForceMetadataRefresh()
 	b.kafkaClient.AddConsumePartitions(map[string]map[int32]kgo.Offset{
 		b.cfg.Kafka.Topic: {
 			partition: kgo.NewOffset().At(startOffset),
 		},
 	})
 	defer b.kafkaClient.RemoveConsumePartitions(map[string][]int32{b.cfg.Kafka.Topic: {partition}})
```

Comment:
Even with the concurrent fetcher, I encountered dodgy behavior/panics/errors when I didn't have these here.

Reply:
I've found that I don't need to call it. Because `AddConsumePartitions` does the same, and I call that right after.